### ISSUE TITLE
[java] Consecutive Literal Appends

### ISSUE THREAD

**Issue by suruchidawar** [pmd/pmd#546]
Consecutive Literal Appends

StringBuffer (or StringBuilder).append is called {0} consecutive times with literal Strings. Use a single append with a single combined String."

```java
sbuffer.append(MailConfigFactory.instance.getHeaderConfig("DEFAULT_HTML_START_AND_CSS_HEADER").getText())
    		.append("<html><table><tr><td class=content>Dear "+objOutputData.getFsUserData().getFirstName()+",</td></tr><br>")
    		.append("<tr><td class=content>Your leave request from "+DateFormatUtility.format(
    				objOutputData.getStartDate(), GlobalConstants.DATE_FORMAT)+" "+DateFormatUtility.format(
    						objOutputData.getStartTime(), GlobalConstants.DISPLAY_TIME_FORMAT)+" to "+DateFormatUtility.format(
    								objOutputData.getEndDate(), GlobalConstants.DATE_FORMAT)+" "+DateFormatUtility.format(
    										objOutputData.getEndTime(), GlobalConstants.DISPLAY_TIME_FORMAT))
    		.append(" "+text +".</td></tr><br>")
    		.append(MailConfigFactory.instance.getFooterConfig("DEFAULT_HTML_TAGS_END_FOOTER").getText())
    		.append("<br><br>")
    		.append(MailConfigFactory.instance.getFooterConfig("DEFAULT_DO_NOT_REPLY_AND_RECEIVE_IN_ERROR_FOOTER").getText());
```

Can you tell why is this reported as a violation since these are not literals which are used in append statement ?

**Comment by jsotuyod** [comment]
The rule checks if there are string literals as part of consecutive calls to append, which imply a suboptimal organization of your append calls.

In your particular case, you are creating a bunch of unnecessary string literals explicitly (ie: `... ",</td></tr><br>").append("<tr><td class=content>...`) and implicitly by string concatenation.

You should consider rewriting your code as:

```java
sbuffer.append(MailConfigFactory.instance.getHeaderConfig("DEFAULT_HTML_START_AND_CSS_HEADER").getText())
    		.append("<html><table><tr><td class=content>Dear ").append(objOutputData.getFsUserData().getFirstName())
                .append(",</td></tr><br><tr><td class=content>Your leave request from ")
                .append(DateFormatUtility.format(objOutputData.getStartDate(), GlobalConstants.DATE_FORMAT))
                .append(' ').append(DateFormatUtility.format(objOutputData.getStartTime(), GlobalConstants.DISPLAY_TIME_FORMAT))
                .append(" to ").append(DateFormatUtility.format(objOutputData.getEndDate(), GlobalConstants.DATE_FORMAT))
                .append(' ').append(DateFormatUtility.format(objOutputData.getEndTime(), GlobalConstants.DISPLAY_TIME_FORMAT))
    		.append(' ').append(text).append(".</td></tr><br>")
    		.append(MailConfigFactory.instance.getFooterConfig("DEFAULT_HTML_TAGS_END_FOOTER").getText())
    		.append("<br><br>")
    		.append(MailConfigFactory.instance.getFooterConfig("DEFAULT_DO_NOT_REPLY_AND_RECEIVE_IN_ERROR_FOOTER").getText());
```

Which does not create unnecessary strings and keep the number of append calls optimal for your given template.
